### ISSUE TITLE
show-build-deps does not strip trailing `-` version separator when using `-E` for hostmakedepends that exist in local repo

### ISSUE THREAD

**Issue by Johnnynator** [void-linux/void-packages#11740]
### System

* xuname:  
  *output of ``xuname`` (part of xtools)*
* package:  
  *affected package(s) including the version*

show-build-deps does not strip trailing `-` version separator when using `-E` for hostmakedepends that exist in local repo, this might also be an issue for makedepends but I haven't tested anything else, just sharing my observations. 

### Expected behavior
```
./xbps-src -m masterdir-musl -a armv7l show-build-deps mailimporter
cmake
extra-cmake-modules
kcoreaddons
libkdepim
python
qt5
```
### Actual behavior
```
./xbps-src -E -m masterdir-musl -a armv7l show-build-deps mailimporter
/media/SourceCode/void-packages/common/xbps-src/shutils/build_dependencies.sh: line 169: unset: XBPS_BINPKG_EXISTS: cannot unset: readonly variable
/media/SourceCode/void-packages/common/xbps-src/shutils/build_dependencies.sh: line 169: unset: XBPS_BINPKG_EXISTS: cannot unset: readonly variable
/media/SourceCode/void-packages/common/xbps-src/shutils/build_dependencies.sh: line 169: unset: XBPS_BINPKG_EXISTS: cannot unset: readonly variable
/media/SourceCode/void-packages/common/xbps-src/shutils/build_dependencies.sh: line 169: unset: XBPS_BINPKG_EXISTS: cannot unset: readonly variable
/media/SourceCode/void-packages/common/xbps-src/shutils/build_dependencies.sh: line 169: unset: XBPS_BINPKG_EXISTS: cannot unset: readonly variable
/media/SourceCode/void-packages/common/xbps-src/shutils/build_dependencies.sh: line 169: unset: XBPS_BINPKG_EXISTS: cannot unset: readonly variable
/media/SourceCode/void-packages/common/xbps-src/shutils/build_dependencies.sh: line 169: unset: XBPS_BINPKG_EXISTS: cannot unset: readonly variable
cmake
extra-cmake-modules-
kcoreaddons-
libkdepim
python
qt5
```
### Steps to reproduce the behavior
TODO: explain a simple case to reproduce it :)

**Comment by CameronNemo** [comment]
Should also affect `show-makedepends` and `show-hostmakedepends`.

**Comment by xtraeme** [comment]
This was fixed by #11874
